EIPS/eip-3332.md

```diff
@@ -0,0 +1,95 @@
+---
+eip: 3332
+title: MEDGASPRICE Opcode
+author: Justice Hudson (@jchancehud)
+discussions-to: https://ethereum-magicians.org/t/medgasprice-opcode-eip/5480
+status: Draft
+type: Standards Track
+category: Core
+created: 2021-03-05
+---
+
+## Simple Summary
+
+Add `MEDGASPRICE (0x46)` that allows smart contracts to access information about current gas prices.
+
+## Abstract
+
+There is an existing opcode `GASPRICE (0x3a)` that allows contracts to access the gas price of the current transaction. This value can be used to impose a maximum gas price on contract calls by setting a constant value, but cannot be used as a long term solution given the fluctuation of gas prices over time. The proposed opcode `MEDGASPRICE (0x46)` would return the median gas price for the parent block. This would allow the implementation of upper bounds on transaction gas prices as a function of the last block median gas price.
+
+## Motivation
+
+With the emergence of rollups as core mechanisms in scaling Ethereum there are a number of common transactions that can be front-run. Optimistic rollups rely on the submission of fraud proofs to maintain the integrity of their systems. As a result actors submitting fraud proofs typically receive a financial reward for doing so. This opens a trivial front-running strategy of watching the mempool for fraud proof submissions and copying such transactions with a much higher gas price to reap the reward. Such front-runners do not perform validation independently and de-incentivize others from performing validation. Adding a mechanism enforcing an upper bound on gas prices for a transaction could be an effective defense against such front-running attacks.
+
+## Specification
+
+If `block.number >= TBD`, add a new opcode:
+
+MEDGASPRICE: 0x46
+
+Pushes the median gas price of the parent block onto the stack.<br />
```

Comment:
```suggestion
Pushes the median gas price of the parent block onto the stack.

```
Try to avoid using HTML in markdown files as it may not be supported in all rendering engines and doesn't read well when read as a raw file.